packages/mui-system/src/Stack/createStack.tsx

```diff
@@ -135,8 +135,12 @@ export const style = ({ ownerState, theme }: StyleFunctionProps) => {
         return { gap: getValue(transformer, propValue) };
       }
       return {
-        '& > :not(style) ~ :not(style)': {
+        // The useFlexGap={false} implement relies on each child to give up control of the margin.
+        // We need to reset the margin to avoid double spacing.
```

Comment:
```suggestion
        // The useFlexGap={false} implementation relies on each child to give up control of the margin.
        // We need to reset the margin to avoid double spacing.
        // TODO: remove this reset https://github.com/mui/material-ui/issues/28035#issuecomment-1615855826
```